### PR TITLE
add more traits to receipts

### PR DESCRIPTION
**Pull request by waynehoover**
- adds trait metadata to the reciept renderer

Todo: tests

### CODE REVIEW

File: contracts/Quest.sol [hunk 1]
```diff
@@ -136,6 +136,16 @@ contract Quest is Ownable, IQuest {
...
     }
 
+    /// @dev Returns the reward amount
+    function getRewardAmount() public view returns (uint256) {
+        return rewardAmountInWeiOrTokenId;
+    }
+
+    /// @dev Returns the reward token address
+    function getRewardToken() public view returns (address) {
+        return rewardToken;
```
Comment (GarrettJMU) [inline contracts/Quest.sol:146]: same note here

File: contracts/RabbitHoleReceipt.sol [hunk 2]
```diff
@@ -69,6 +72,12 @@ contract RabbitHoleReceipt is
         royaltyRecipient = royaltyRecipient_;
     }
 
+    /// @dev set the quest factory contract
+    /// @param questFactory_ the address of the quest factory contract
+    function setQuestFactory(address questFactory_) public onlyOwner {
+        QuestFactoryContract = IQuestFactory(questFactory_);
```
Comment (GarrettJMU) [inline contracts/RabbitHoleReceipt.sol:78]: Is it more common to use Interface here or the actual contract?

File: contracts/Quest.sol [hunk 3]
```diff
@@ -136,6 +136,16 @@ contract Quest is Ownable, IQuest {
         return claimedList[tokenId_] == true;
     }
 
+    /// @dev Returns the reward amount
+    function getRewardAmount() public view returns (uint256) {
+        return rewardAmountInWeiOrTokenId;
```
[the remark this answers is not in the document]
Comment (waynehoover) [reply contracts/Quest.sol:141]: yes, but I'm not sure how to access it via the interface only. If I include the contract we get a circular dependency and it won't compile.

File: contracts/QuestFactory.sol [hunk 4]
```diff
@@ -170,6 +170,16 @@ contract QuestFactory is Initializable, OwnableUpgradeable, AccessControlUpgrade
...
     }
 
+    /// @dev return data in the quest struct for a questId
+    /// @param questId_ The id of the quest
+    function questInfo(string memory questId_) external view returns (address, uint, uint) {
+        return (
+            quests[questId_].questAddress,
+            quests[questId_].totalParticipants,
+            quests[questId_].numberMinted
+        );
```
Comment (jonathandiep) [inline contracts/QuestFactory.sol:180]: I like this! But haven't really worked with tuples on the ethers.js side so interesting to see how the data will return (maybe a JS object?)

File: contracts/RabbitHoleReceipt.sol (hunk 2 above)
Comment (waynehoover) [reply contracts/RabbitHoleReceipt.sol:78]: the actual contract would have been better, but it creates a circular dependency and I couldn't.

File: contracts/QuestFactory.sol (hunk 4 above)
Comment (jonathandiep) [reply contracts/QuestFactory.sol:180]: Nvm I see this mostly for RabbitHoleReceipt.sol

**Review by jonathandiep — APPROVED** [review]
lgtm

**Pull request merged**